### ISSUE TITLE
Add compare-and-swap fields in SetLien request

### ISSUE THREAD

**Issue by dckc** [Agoric/Agoric#4887]
## What is the Problem Being Solved?

We need to keep lien state in sync between JS and golang data structures. The `setLien` API with just the `target` amount seems to have lost-update risks.

## Description of the Design

Add another denom / amount pair of args to the current message structure:

```
{
        type: XLien.LIEN_GET_ACCOUNT_STATE,
        address,
        denom: XLien.denom,
        amount: '0',
      }
```

## Security Considerations

n/a

## Test Plan

usual unit testing

**Comment by dckc** [comment]
discussing this with @JimLarson ... sounds like delta in, new state out is the way to go

**Comment by JimLarson** [comment]
Adding some comments on underlying race and the delta-based design vs compare-and-swap.

The attestation contract maintains a non-authoritative record of liens and uses this to compute the new desired lien amount and sends a message over the bridge to the Cosmos layer to set the (authoritative) liened amount to that value. When the Cosmos layer returns, the contract updates its non-authoritative record to the calculated new amount.

The problem is that the message round-trip is not synchronous. Multiple lien change requests can come in while one is in flight, and their calculated new state will be wrong.

The contract's record must be non-authoritative since lien changes race with staking changes at the Cosmos level.  It's not simple for the contract to modify it's record ahead of sending the request to the Cosmos layer, since the request might fail. The simplest solution is to abandon simply commanding a new state.

The compare-and-swap approach will detect harmful pipelining and reject later pipelined operations, but this is unnecessary since pipelined delta-based requests are perfectly fine. The operations on the contract are fundamentally delta-based - there is a request to either increase or decrease the lien by a given amount. Calculating the new liened amount is redundant and unnecessary.

I *think* that using a delta-based protocol makes the contract-side non-authoritative store unnecessary, for potential space savings.

**Comment by JimLarson** [comment]
Closed by #5111
